python/ray/tests/test_resource_demand_scheduler.py

```diff
@@ -226,6 +228,46 @@ def testRequestBundles(self):
         assert self.provider.mock_nodes[2].instance_type == "m4.16xlarge"
         assert self.provider.mock_nodes[3].instance_type == "m4.16xlarge"
 
+    def testResourcePassing(self):
+        config = MULTI_WORKER_CLUSTER.copy()
+        config["min_workers"] = 0
+        config["max_workers"] = 50
+        config_path = self.write_config(config)
+        self.provider = MockProvider(default_instance_type="m4.large")
+        runner = MockProcessRunner()
+        autoscaler = StandardAutoscaler(
+            config_path,
+            LoadMetrics(),
+            max_failures=0,
+            process_runner=runner,
+            update_interval_s=0)
+        assert len(self.provider.non_terminated_nodes({})) == 0
+        autoscaler.update()
+        self.waitForNodes(0)
+        autoscaler.request_resources([{"CPU": 1}])
+        autoscaler.update()
+        self.waitForNodes(1)
+        assert self.provider.mock_nodes[0].instance_type == "m4.large"
+        autoscaler.request_resources([{"GPU": 8}])
+        autoscaler.update()
+        self.waitForNodes(2)
+        assert self.provider.mock_nodes[1].instance_type == "p2.8xlarge"
+
+        # TODO (Alex): Autoscaler creates the node during one update then
+        # starts the updater in the enxt update. The sleep is largely
+        # unavoidable because the updater runs in its own thread and we have no
+        # good way of ensuring that the commands are sent in time.
+        autoscaler.update()
+        sleep(0.1)
```

Comment:
Note this is isn't the only use of sleep, the rest are all just hidden behind `MockProvider.waitForNodes` (therefore it's not more flaky than the rest of the tests, just equally flaky :p). 

I'll remove the uses of sleep in a future PR